cmd/conformance/gcp/kmsnote/main.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// kmsnote is a tool for creating a note verifier from an Ed25519 public key
+// retrieved from KMS.
+package main
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+
+	kms "cloud.google.com/go/kms/apiv1"
+	"golang.org/x/mod/sumdb/note"
+
+	"cloud.google.com/go/kms/apiv1/kmspb"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// KeyVersionNameFormat is the GCP resource identifier for a key version.
+	// google.cloud.kms.v1.CryptoKeyVersion.name
+	// https://cloud.google.com/kms/docs/reference/rest/v1/projects.locations.keyRings.cryptoKeys.cryptoKeyVersions
+	KeyVersionNameFormat = "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s/cryptoKeyVersions/%d"
+)
+
+var (
+	keyID = flag.String("key_id", "", "cryptoKeyVersion ID string ('projects/.../locations/.../keyRings/.../cryptoKeys/.../cryptoKeyVersions/...')")
+	name  = flag.String("name", "", "Name for generated note Verifier")
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+
+	ctx := context.Background()
+	kmClient, err := kms.NewKeyManagementClient(ctx)
+	if err != nil {
+		log.Fatalf("failed to create KeyManagementClient: %v", err)
+	}
+	defer kmClient.Close()
+
+	v, err := verifierKeyString(ctx, kmClient, *keyID, *name)
+	if err != nil {
+		klog.Exitf("Failed to generate verifier string: %v", err)
+	}
+
+	fmt.Println(v)
```

Comment:
Can this line be removed? Is this for debug use?

Reply:
This is the most important line in the tool - this prints out the note verifier string we've just created :)

Reply:
Ah right, should we use `klog` instead of `fmt.Println()`?

Reply:
No - I wanted it to go to `stdout` without any decoration so that we can e.g. set ENV vars with it (`export VERIFIER=$(go run ./cmd/conformance/gcp/kmsnote ...)`), etc.

Reply:
Gotcha, it makes sense to me. It would be great if you can put this use case into the package doc comment.